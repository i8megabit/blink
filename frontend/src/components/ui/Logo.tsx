import React from 'react';

interface LogoProps {
  size?: 'sm' | 'md' | 'lg' | 'xl';
  variant?: 'default' | 'white' | 'dark';
  className?: string;
  showText?: boolean;
}

const Logo: React.FC<LogoProps> = ({ 
  size = 'md', 
  variant = 'default', 
  className = '',
  showText = true 
}) => {
  const sizeClasses = {
    sm: 'w-16 h-8',
    md: 'w-24 h-10',
    lg: 'w-32 h-12',
    xl: 'w-40 h-16',
  };

  const variantClasses = {
    default: 'text-gray-900',
    white: 'text-white',
    dark: 'text-gray-100',
  };

  const iconSize = {
    sm: 16,
    md: 20,
    lg: 24,
    xl: 32,
  };

  const textSize = {
    sm: 'text-sm',
    md: 'text-base',
    lg: 'text-lg',
    xl: 'text-xl',
  };

  return (
    <div className={`flex items-center gap-2 ${sizeClasses[size]} ${className}`}>
      {/* Иконка */}
      <div className="relative">
        <svg 
          width={iconSize[size]} 
          height={iconSize[size]} 
          viewBox="0 0 24 24" 
          className="flex-shrink-0"
        >
          <defs>
            <linearGradient id="logoGradient" x1="0%" y1="0%" x2="100%" y2="100%">
              <stop offset="0%" style={{ stopColor: '#0ea5e9', stopOpacity: 1 }} />
              <stop offset="100%" style={{ stopColor: '#0284c7', stopOpacity: 1 }} />
            </linearGradient>
            <filter id="logoShadow" x="-20%" y="-20%" width="140%" height="140%">
              <feDropShadow dx="0" dy="1" stdDeviation="2" floodColor="#000000" floodOpacity="0.1"/>
            </filter>
          </defs>
          
          {/* Фоновая форма */}
          <circle 
            cx="12" 
            cy="12" 
            r="10" 
            fill="url(#logoGradient)" 
            filter="url(#logoShadow)"
          />
          
          {/* Иконка ссылки */}
          <g transform="translate(4, 4)">
            <path 
              d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z" 
              fill="white" 
              opacity="0.9"
            />
            <path 
              d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z" 
              fill="white" 
              opacity="0.9"
            />
            <path 
              d="M12 6l-2 2 2 2" 
              stroke="white" 
              strokeWidth="1.5" 
              fill="none" 
              strokeLinecap="round"
            />
          </g>
        </svg>
      </div>

      {/* Текст */}
      {showText && (
        <span className={`font-semibold ${textSize[size]} ${variantClasses[variant]} tracking-tight`}>
          reLink
        </span>
      )}
    </div>
  );
};

export default Logo; 