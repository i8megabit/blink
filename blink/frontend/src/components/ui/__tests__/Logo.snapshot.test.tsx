import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Logo from '../Logo';

describe('Logo Snapshot Tests', () => {
  it('renders default logo correctly', () => {
    const { container } = render(<Logo />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders logo without text correctly', () => {
    const { container } = render(<Logo showText={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders small logo correctly', () => {
    const { container } = render(<Logo size="sm" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders large logo correctly', () => {
    const { container } = render(<Logo size="lg" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders extra large logo correctly', () => {
    const { container } = render(<Logo size="xl" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders white variant correctly', () => {
    const { container } = render(<Logo variant="white" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders dark variant correctly', () => {
    const { container } = render(<Logo variant="dark" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with custom className correctly', () => {
    const { container } = render(<Logo className="custom-class" />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 