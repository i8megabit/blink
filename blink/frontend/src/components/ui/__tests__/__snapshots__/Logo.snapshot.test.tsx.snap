// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo Snapshot Tests > renders dark variant correctly 1`] = `
<div
  class="flex items-center gap-2 w-24 h-10 "
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="20"
      viewBox="0 0 24 24"
      width="20"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
  <span
    class="font-semibold text-base text-gray-100 tracking-tight"
  >
    Relink
  </span>
</div>
`;

exports[`Logo Snapshot Tests > renders default logo correctly 1`] = `
<div
  class="flex items-center gap-2 w-24 h-10 "
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="20"
      viewBox="0 0 24 24"
      width="20"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
  <span
    class="font-semibold text-base text-gray-900 tracking-tight"
  >
    Relink
  </span>
</div>
`;

exports[`Logo Snapshot Tests > renders extra large logo correctly 1`] = `
<div
  class="flex items-center gap-2 w-40 h-16 "
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="32"
      viewBox="0 0 24 24"
      width="32"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
  <span
    class="font-semibold text-xl text-gray-900 tracking-tight"
  >
    Relink
  </span>
</div>
`;

exports[`Logo Snapshot Tests > renders large logo correctly 1`] = `
<div
  class="flex items-center gap-2 w-32 h-12 "
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
  <span
    class="font-semibold text-lg text-gray-900 tracking-tight"
  >
    Relink
  </span>
</div>
`;

exports[`Logo Snapshot Tests > renders logo without text correctly 1`] = `
<div
  class="flex items-center gap-2 w-24 h-10 "
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="20"
      viewBox="0 0 24 24"
      width="20"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
</div>
`;

exports[`Logo Snapshot Tests > renders small logo correctly 1`] = `
<div
  class="flex items-center gap-2 w-16 h-8 "
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="16"
      viewBox="0 0 24 24"
      width="16"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
  <span
    class="font-semibold text-sm text-gray-900 tracking-tight"
  >
    Relink
  </span>
</div>
`;

exports[`Logo Snapshot Tests > renders white variant correctly 1`] = `
<div
  class="flex items-center gap-2 w-24 h-10 "
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="20"
      viewBox="0 0 24 24"
      width="20"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
  <span
    class="font-semibold text-base text-white tracking-tight"
  >
    Relink
  </span>
</div>
`;

exports[`Logo Snapshot Tests > renders with custom className correctly 1`] = `
<div
  class="flex items-center gap-2 w-24 h-10 custom-class"
>
  <div
    class="relative"
  >
    <svg
      class="flex-shrink-0"
      height="20"
      viewBox="0 0 24 24"
      width="20"
    >
      <defs>
        <lineargradient
          id="logoGradient"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            style="stop-color: rgb(14, 165, 233); stop-opacity: 1;"
          />
          <stop
            offset="100%"
            style="stop-color: rgb(2, 132, 199); stop-opacity: 1;"
          />
        </lineargradient>
        <filter
          height="140%"
          id="logoShadow"
          width="140%"
          x="-20%"
          y="-20%"
        >
          <fedropshadow
            dx="0"
            dy="1"
            flood-color="#000000"
            flood-opacity="0.1"
            stdDeviation="2"
          />
        </filter>
      </defs>
      <circle
        cx="12"
        cy="12"
        fill="url(#logoGradient)"
        filter="url(#logoShadow)"
        r="10"
      />
      <g
        transform="translate(4, 4)"
      >
        <path
          d="M8 4C5.79 4 4 5.79 4 8s1.79 4 4 4h2v-2H8c-1.1 0-2-.9-2-2s.9-2 2-2h2V4H8z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M16 4h-2v2h2c1.1 0 2 .9 2 2s-.9 2-2 2h-2v2h2c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
          fill="white"
          opacity="0.9"
        />
        <path
          d="M12 6l-2 2 2 2"
          fill="none"
          stroke="white"
          stroke-linecap="round"
          stroke-width="1.5"
        />
      </g>
    </svg>
  </div>
  <span
    class="font-semibold text-base text-gray-900 tracking-tight"
  >
    Relink
  </span>
</div>
`;
